test/e2e/redundancy/route_agent_failover.go

```diff
@@ -0,0 +1,76 @@
+/*
+© 2021 Red Hat, Inc. and others
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package redundancy
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	"github.com/submariner-io/shipyard/test/e2e/tcp"
+
+	subFramework "github.com/submariner-io/submariner/test/e2e/framework"
+)
+
+var _ = Describe("[redundancy] Route Agent restart tests", func() {
+	f := subFramework.NewFramework("route-agent-restart")
+
+	When("a Route Agent pod is restarted on the Gateway node", func() {
+		It("should start a new Route Agent pod and be able to connect from another cluster", func() {
+			testRouteAgentRestartOnGateway(f)
+		})
+	})
+
+})
+
+func testRouteAgentRestartOnGateway(f *subFramework.Framework) {
+	clusterAName := framework.TestContext.ClusterIDs[framework.ClusterA]
+	clusterBName := framework.TestContext.ClusterIDs[framework.ClusterB]
+
+	gatewayNodes := f.FindNodesByGatewayLabel(framework.ClusterA, true)
+	Expect(gatewayNodes).To(HaveLen(1), fmt.Sprintf("Expected only one gateway node on %q", clusterAName))
+	By(fmt.Sprintf("Found gateway on node %q on %q", gatewayNodes[0].Name, clusterAName))
+
+	routeAgentPod := f.AwaitRouteAgentPod(framework.ClusterA)
+	By(fmt.Sprintf("Found route agent pod %q on %q", routeAgentPod.Name, clusterAName))
+
+	By(fmt.Sprintf("Deleting route agent pod %q", routeAgentPod.Name))
+	f.DeletePod(framework.ClusterA, routeAgentPod.Name, framework.TestContext.SubmarinerNamespace)
+
+	newRouteAgentPod := f.AwaitRouteAgentPod(framework.ClusterA)
```

Comment:
1) Here , there is no need for another method:
https://github.com/submariner-io/submariner/pull/1061#discussion_r551628916
Suggested change
        NewRouteAgentPod := f.AwaitRouteAgentPod(framework.ClusterA) ->
        newRouteAgentPod := f.AwaitNewRouteAgentPod(framework.ClusterA)

since I can resuse the f.AwaitRouteAgentPod and put it in a newRouteAgentPod variable.

Reply:
Please see `AwaitNewSubmarinerEnginePod` and how it is used in the other test.

Reply:
where is AwaitRouteAgentPod defined?

Does it wait for a route agent on any node?, or only the Gateway node?

If it's only in the gateway node, please rename to AwaitRouteAgentPodOnGatewayNode, or make
a function that receives the worker node name: AwaitRouteAgentPodOn(nodeName), as deployments will have as many route agents as nodes.